src/csvcubed/writers/helpers/qbwriter/dsdtordfmodelshelper.py

```diff
@@ -301,17 +301,29 @@ def _get_qb_component_specs_for_col(
                 self._get_qb_attribute_specification(column_name_uri_safe, component)
             ]
         elif isinstance(component, QbMultiUnits):
-            return [self._get_qb_units_column_specification(column_name_uri_safe)]
+            return self._get_qb_units_column_specification(column_name_uri_safe)
         elif isinstance(component, QbMultiMeasureDimension):
             return self._get_qb_measure_dimension_specifications(component)
         elif isinstance(component, QbObservationValue):
             return self._get_qb_obs_val_specifications(component)
         else:
             raise TypeError(f"Unhandled component type {type(component)}")
 
+    _units_component_already_defined: bool = False
+    """
+    Records whether or not a units component has already been defined in this cube.
+    If it has, don't define it again.
+    """
+
     def _get_qb_units_column_specification(
         self, column_name_uri_safe: str
-    ) -> rdf.qb.AttributeComponentSpecification:
+    ) -> List[rdf.qb.AttributeComponentSpecification]:
+        if self._units_component_already_defined:
+            # Don't define a second units component, the first one will work just fine.
+            return []
```

Comment:
Can you add some debug logging here so we can track that we're avoiding serialising a second units component?

Reply:
```    def _get_qb_units_column_specification(
        self, column_name_uri_safe: str
    ) -> List[rdf.qb.AttributeComponentSpecification]:
        if self._units_component_already_defined:
            _logger.debug(
                "The component type has already been serialised, %s.",
                column_name_uri_safe,
            )
            # Don't define a second units component, the first one will work just fine.

            return []```

is this the right place for it?

Reply:
that's definitely the right place, but could it say something like `Units component already generated. Not generating a second one.`

Reply:
Yeah cool I'll sort it out